ftp/src/main/scala/akka/stream/alpakka/ftp/model.scala

```diff
@@ -129,3 +130,37 @@ object FtpCredentials {
    */
   final case class NonAnonFtpCredentials(username: String, password: String) extends FtpCredentials
 }
+
+object SftpIdentity {
+
+  /** Java API */
+  def createEmptyIdentity: Option[SftpIdentity] = None
+
+  /** Java API */
+  def createSftpIdentity(name: String, privateKey: Array[Byte]): Option[SftpIdentity] =
+    Some(SftpIdentity(name, privateKey))
+
+  /** Java API */
+  def createSftpIdentity(name: String, privateKey: Array[Byte], password: Array[Byte]): Option[SftpIdentity] =
+    Some(SftpIdentity(name, privateKey, Some(password)))
+
+  /** Java API */
+  def createSftpIdentity(name: String,
+                         privateKey: Array[Byte],
+                         password: Array[Byte],
+                         publicKey: Array[Byte]): Option[SftpIdentity] =
+    Some(SftpIdentity(name, privateKey, Some(password), Some(publicKey)))
+}
+
+/**
+ * SFTP identity for authenticating using private/public key
+ *
+ * @param name name of identity
+ * @param privateKey private key to use when connecting
+ * @param password password to use to decrypt private key
+ * @param publicKey public key to use when connecting
+ */
+case class SftpIdentity(name: String,
+                        privateKey: Array[Byte],
+                        password: Option[Array[Byte]] = None,
+                        publicKey: Option[Array[Byte]] = None)
```

Comment:
I wonder if it would be interesting to offer variants of the identity specification based on files on the classpath in addition to these `Array[Byte]`.